trinity/plugins/builtin/peer_discovery/plugin.py

```diff
@@ -0,0 +1,143 @@
+from argparse import (
+    ArgumentParser,
+    _SubParsersAction,
+)
+import asyncio
+
+from cancel_token import (
+    CancelToken,
+)
+from eth_typing import (
+    BlockNumber,
+)
+
+from eth.constants import (
+    GENESIS_BLOCK_NUMBER
+)
+
+from p2p.discovery import (
+    get_v5_topic,
+    DiscoveryByTopicProtocol,
+    PreferredNodeDiscoveryProtocol,
+)
+from p2p.kademlia import (
+    Address,
+)
+from p2p.protocol import (
+    Protocol,
+)
+
+from trinity.config import (
+    TrinityBeaconConfig,
+    TrinityConfig,
+)
+from trinity.db.manager import (
+    create_db_manager
+)
+from trinity.extensibility import (
+    BaseIsolatedPlugin,
+)
+from trinity.protocol.bcc.proto import (
+    BCCProtocol,
+)
+from trinity.protocol.eth.proto import (
+    ETHProtocol,
+)
+from trinity.protocol.les.proto import (
+    LESProtocolV2,
+)
+from trinity.utils.shutdown import (
+    exit_with_service_and_endpoint,
+)
+from .discovery_service import (
+    DiscoveryService,
+)
+
+
+def get_protocol(trinity_config: TrinityConfig) -> Protocol:
+    # For now DiscoveryByTopicProtocol supports a single topic, so we use the latest
+    # version of our supported protocols. Maybe this could be more generic?
+    if isinstance(trinity_config, TrinityBeaconConfig):
+        return BCCProtocol
```

Comment:
@jannikluhn @hwwhww This just got merged. In theory this means that the `trinity-beacon` node can use the existing discovery now (including discv5). This is, because this plugin is currently provided as a base plugin for `trinity` as well as `trinity-beacon` (https://github.com/ethereum/py-evm/pull/1556/files#diff-bcb3f875bb745a964ec62cfb0d9ac188R44)

In practice though, the `trinity-beacon` command is just a stub at this point that just spins up and then kills itself right after bootstrap. Just pointing this out to give you a sense of how things are starting to get wired together.